src/components/pill-selector-dropdown/PillSelectorDropdownField.js

```diff
@@ -0,0 +1,127 @@
+// @flow
+import * as React from 'react';
+import getProp from 'lodash/get';
+import isEqual from 'lodash/isEqual';
+
+import type { FieldProps } from 'formik';
+
+import DatalistItem from '../datalist-item';
+import PillSelectorDropdown from './PillSelectorDropdown';
+import type { Option, Pill } from './flowTypes';
+
+import './PillSelectorDropdown.scss';
+
+type Props = FieldProps & {
+    /** CSS class for the component */
+    className?: string,
+    /** If true, the input control is disabled so no more input can be made */
+    disabled?: boolean,
+    /** If true, user can add pills not included in selector options */
+    isCustomInputAllowed: boolean,
+    /** Input label */
+    label: React.Node,
+    /** Array or Immutable list with data for the selected options shown as pills */
+    options: Array<Option>,
+    /** A placeholder to show in the input when there are no pills */
+    placeholder: string,
+    /** Called to check if pill item data is valid. The `item` is passed in. */
+    validator: (text: string) => boolean,
+};
+
+type State = {
+    selectorOptions: Array<Option>,
+};
+
+class PillSelectorDropdownField extends React.PureComponent<Props, State> {
+    static defaultProps = {
+        isCustomInputAllowed: false,
+        options: [],
+        validator: () => true,
+    };
+
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            selectorOptions: props.options,
+        };
+    }
+
+    deNormalize(options: Array<Option>): Array<Pill> {
+        return options.map((option: Option) => ({
+            text: option.displayText,
+            value: option.value,
+        }));
+    }
+
+    normalize(pills: Array<Pill>): Array<Option> {
+        return pills.map((pill: Pill) => ({
+            displayText: pill.text,
+            value: pill.value,
+        }));
+    }
+
+    createFakeSyntheticEvent(name: string, value: Array<Option>) {
```

Comment:
Is this returning the full shape of a synthetic event? If so, can we add it as a return type? If not, could we just call this `this.handleChange` and leave a comment describing the need for `currentTarget` and `target?

Reply:
Also, for the situation where the handler is debounced or something, will this work with the .persist() method on React synthetic events?

Reply:
I can rename. I dont have access to the original event, for that will have to refactor the actual pill selector. Also not sure what target it should be anyways even if we did that. Since the "target" for this component is not really the input you type in, but this whole custom component itself. Formik is fine getting a fake event target such as this, and it will not call persist() if that function does not exist.

Alternative to calling onChange with fake event target, would be to use Formik's setFieldValue, but I am trying to keep all these formik wrappers free off formik specific functions.